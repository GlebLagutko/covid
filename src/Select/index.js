import React, {useEffect, useState} from "react";

function Select({country, setCountry}) {
    let [countries, setCountries] = useState([]);


    function fetchCountries() {
        fetch('https://disease.sh/v3/covid-19/jhucsse').then(response => response.json()).then(data => {
            let arr = [];
            data.forEach(el => arr.push(el.country));

            arr = [...new Set(arr)];

            countries = [...arr];
            setCountries(countries);

        });
    }

    useEffect(() => {
        fetchCountries();
    }, []);

    const elems = [<option value={'All'} key={'All'}>All</option>];
    countries.forEach(elem => elems.push(<option value={elem} key={elem}>{elem}</option>))


    return (
        <select value={country} onChange={e => setCountry(e.target.value)}>
            {
                elems.map(el => el)
            }
        </select>
    );
}

export default Select;