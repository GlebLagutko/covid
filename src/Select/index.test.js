import {render} from '@testing-library/react';

import Select from '.';
import React from "react";

describe('Select', () => {

  it('renders correctly', async () => {

    const {container} = render(<Select/>);

    expect(container).toMatchSnapshot();
  });
});