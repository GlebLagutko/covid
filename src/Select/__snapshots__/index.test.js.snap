// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<div>
  <select>
    <option
      value="All"
    >
      All
    </option>
  </select>
</div>
`;
